### PR TITLE
Valid url

### PR DESCRIPTION
**Pull request by stevehipwell**
case 1. Add a new type definition.
- [x ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

File: valid-url/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (vvakame) [inline valid-url/tsconfig.json]: could you try to use `"strictNullChecks": true,`?

**Comment by vvakame** [comment]
*valid-url/valid-url.d.ts*

Checklist

* [ ] is correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/valid-url
  * http://bower.io/search/?q=valid-url
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (valid-url-tests.ts or valid-url-tests.tsx)
* [ ] pass the Travis CI test?

File: valid-url/tsconfig.json (hunk 1 above)
Comment (stevehipwell) [inline valid-url/tsconfig.json]: I've updated this, but I'm not sure why it needs doing (all examples I looked at have it set to false)?

File: valid-url/valid-url.d.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for valid-url v1.0.9
+// Project: https://github.com/ogt/valid-url
+// Definitions by: Steve Hipwell <https://github.com/stevehipwell>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'valid-url' {
```
Comment (vvakame) [inline valid-url/valid-url.d.ts]: https://github.com/Microsoft/TypeScript/issues/7125
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/types-2.0/esprima/index.d.ts#L8-L9

**Comment by minestarks** [comment]
@stevehipwell Sorry for leaving this PR open for a long time -- I would merge it but now it seems there's a merge conflict. Could you resolve it? Thank you.

**Comment by paulvanbrenk** [comment]
This appears to be already merged in from `master`

**Pull request closed without merging**